deployment/terraform/modules/docker_host/main.tf

```diff
@@ -0,0 +1,60 @@
+resource "aws_security_group" "allow_docker" {
+  name        = "${var.security_group_name}"
+  description = "Allow connection to docker engine from within VPC"
+  vpc_id      = "${var.vpc_id}"
+
+  # Explicitly set rule for port 2376 since this is the conventional port used
+  # for interacting with docker over tcp
+  ingress {
+    from_port   = 2376
+    to_port     = 2376
+    protocol    = "tcp"
+    cidr_blocks = ["${var.vpc_cidr_block}"]
+  }
+
+  # Allow access on IANA User ports:
+  # https://www.iana.org/assignments/service-names-port-numbers/service-names-port-numbers.xhtml
+  ingress {
+    from_port   = 1024
+    to_port     = 49151
+    protocol    = "tcp"
+    cidr_blocks = ["${var.vpc_cidr_block}"]
+  }
+
+  egress {
+    from_port   = 0             # Implicit with AWS, but Terraform requires that it be explicit:
+    to_port     = 0
+    protocol    = "-1"
+    cidr_blocks = ["0.0.0.0/0"]
+  }
+  tags {
+    Name = "${var.security_group_name}"
+  }
+}
+
+resource "aws_instance" "docker_host" {
+  ami                    = "ami-2757f631"
+  subnet_id              = "${var.private_subnet_id}"
+  instance_type          = "t2.medium"
+  vpc_security_group_ids = ["${aws_security_group.allow_docker.id}"]
+  root_block_device {
+    volume_size = "20"
+  }
+  user_data = <<EOF
+#!/bin/bash
+set -o errexit
+set -o verbose
+sudo apt-get update
+sudo apt -yq install docker.io
+sudo mkdir /lib/systemd/system/docker.service.d
+sudo echo -e '[Service]\nExecStart=\nExecStart=/usr/bin/dockerd -H fd:// -H tcp://0.0.0.0:2376' > /lib/systemd/system/docker.service.d/startup_options.conf
+sudo systemctl daemon-reload
+sudo service docker restart
+EOF
```

Comment:
Would that become a headache if we just want this one thing?

Its easy to determine if this machine has docker installed from the Refinery instance (use docker-py client, launch a tool, etc.), but it would be more difficult to determine whether a puppet manifest applied properly. We'd have to configure ssh access from the refinery host to the docker one to be able to tail logs etc. which I don't think we need to do.

Maybe less is more in this case?

Reply:
The problem is that there are already two different ways to provision the environment: Puppet on AWS/Vagrant and a custom script on Travis CI. Now Docker configuration is applied with Puppet on Vagrant and with a custom script on AWS. This leads to code duplication and can also introduce subtle differences in configuration that could make troubleshooting very difficult.

Also, what makes you think that a shell script is more reliable that applying a Puppet manifest? If Docker provisioning fails then you still have to examine the logs in either case.

Reply:
I was just thinking that we're trying to do something very specific with this 2nd host and the time investment to configure and provision it with puppet may not be worth it at the moment if we have growing pains with this approach and need to move to a more scalable solution.